framework/src/transfers/MultiAppGeneralFieldNearestNodeTransfer.C

```diff
@@ -0,0 +1,213 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MultiAppGeneralFieldNearestNodeTransfer.h"
+
+// MOOSE includes
+#include "DisplacedProblem.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "MooseTypes.h"
+#include "MooseVariableFE.h"
+
+#include "libmesh/generic_projector.h"
+#include "libmesh/meshfree_interpolation.h"
+#include "libmesh/system.h"
+#include "libmesh/mesh_function.h"
+#include "libmesh/mesh_tools.h"
+#include "libmesh/parallel_algebra.h" // for communicator send and receive stuff
+
+// TIMPI includes
+#include "timpi/communicator.h"
+#include "timpi/parallel_sync.h"
+
+registerMooseObject("MooseApp", MultiAppGeneralFieldNearestNodeTransfer);
+
+InputParameters
+MultiAppGeneralFieldNearestNodeTransfer::validParams()
+{
+  InputParameters params = MultiAppGeneralFieldTransfer::validParams();
+  params.addClassDescription(
+      "Transfers field data at the MultiApp position by finding the value at the nearest neighbor "
+      "in the origin application.");
+  params.addParam<unsigned int>("num_nearest_points",
+                                1,
+                                "Number of nearest source (from) points will be chosen to "
+                                "construct a value for the target point.");
+  return params;
+}
+
+MultiAppGeneralFieldNearestNodeTransfer::MultiAppGeneralFieldNearestNodeTransfer(
+    const InputParameters & parameters)
+  : MultiAppGeneralFieldTransfer(parameters),
+    _num_nearest_points(getParam<unsigned int>("num_nearest_points"))
+{
+}
+
+void
+MultiAppGeneralFieldNearestNodeTransfer::prepareEvaluationOfInterpValues(
+    const VariableName & var_name)
+{
+  _local_kdtrees.clear();
```

Comment:
This is a lot of cache that's getting cleared but will need to be recreated the next time this variable gets transferred.  Should we be indexing these caches (and the `MeshFunction`s used elsewhere, etc) by variable name, then only clearing and regenerating them when we *have to* because a mesh changed?

Reply:
We actually do not have the mesh changed interface in transfers so we cant do that right now.
Do you want this to be in scope?

Reply:
Nah, not if it's going to be difficult.  IMHO only getting a *wrong* answer should block a PR; getting a right answer too slowly can be fixed later (or ignored, if profiling doesn't show it matters).